File: src/files/FileSFTPFile.java
package files;

import com.jcraft.jsch.ChannelSftp.LsEntry;

public class FileSFTPFile implements FileCommon {
	
	private final LsEntry file;
	
	public FileSFTPFile(LsEntry file) {
		this.file = file;
	}

	@Override
	public String getFileName() {
		return file.getFilename();
	}

	@Override
	public long getFileSize() {
		return file.getAttrs().getSize();
	}

	@Override
	public boolean isFile() {
		return file.getAttrs().isReg();
	}

	@Override
	public boolean isDirectory() {
		return file.getAttrs().isDir();
	}

}

File: src/files/FileSFTPFile.kt
package files

import com.jcraft.jsch.ChannelSftp.LsEntry

class FileSFTPFile(private val file: LsEntry) : FileCommon {
    override fun getFileName(): String {
        return file.filename
    }

    override fun getFileSize(): Long {
        return file.attrs.size
    }

    override fun isFile(): Boolean {
        return file.attrs.isReg
    }

    override fun isDirectory(): Boolean {
        return file.attrs.isDir
    }
}
